File: dart/resistor-color-duo/lib/resistor_color_duo.dart
class ResistorColorDuo {
  final colorsMapped = const {
    'black': '0',
    'brown': '1',
    'red': '2',
    'orange': '3',
    'yellow': '4',
    'green': '5',
    'blue': '6',
    'violet': '7',
    'grey': '8',
    'white': '9'
  };
  int value(List<String> list) {
    String results = '';
    for (int i = 0; i < 2; i++) {
      results += colorsMapped[list[i]];
    }
    return int.parse(results.trim());
  }
}

File: typescript/resistor-color-duo/resistor-color-duo.ts
const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
] as const;
type Color = typeof COLORS[number];
export class ResistorColor {
  private colors: Color[];
  constructor(colors: Color[]) {
    this.colors = colors;
    if (this.colors.length < 2) {
      throw Error('At least two colors need to be present');
    }
  }
  value(): number {
    const result =
      COLORS.indexOf(this.colors[0]) * 10 + COLORS.indexOf(this.colors[1]);
    return result;
  }
}
